### ISSUE TITLE
Not showing output for SQL query longer than 0.5s - no error, no results, just hides query box

### ISSUE THREAD

**Issue by crisps-for-breakfast** [phpmyadmin/phpmyadmin#18141]
If I run a query through the SQL tab that takes longer than 0.5s to run, I get no output, and no errors, all that happens is that the query box is hidden. Nothing in my JS console, or server error logs `/var/log/httpd/*.log` or `/var/log/php-fpm/*.log`

Looks like it runs the query as the time taken to spin and hide the query box appears consistent with the time to run the query. I should note that longer queries seem to run without issue in the relevant PHP application, and in the MariaDB command line on the same box

- Login and go to SQL tab
- Enter `SELECT SLEEP(0.5)` and hit Go - No Output
- Enter `SELECT SLEEP(0.4)` and hit Go - Output as expected

SELECT SLEEP(0.5) results
![image](https://user-images.githubusercontent.com/43699463/220989469-e90068cb-adce-4ad3-8054-4a6091929fe3.png)

SELECT SLEEP(0.4) results
![image](https://user-images.githubusercontent.com/43699463/220989595-d44a2f79-8efa-440c-8197-fab17643cf7b.png)

Versions
![image](https://user-images.githubusercontent.com/43699463/220990140-72ba1e2b-ab12-4264-ba56-f6cd3d3a2637.png)

Apache: 2.4.54 using PHP-FPM
OS Server: Amazon Linux 2 (Arm 64) 
OS Client: macOS Monterey 12.6.3
Browser: Chrome Version 110.0.5481.177 (Official Build) (x86_64)

Steps taken to install phpMyAdmin on server:

```
cd /usr/share
sudo wget https://www.phpmyadmin.net/downloads/phpMyAdmin-latest-all-languages.tar.gz
sudo mkdir phpMyAdmin
sudo tar -xvzf phpMyAdmin-latest-all-languages.tar.gz -C phpMyAdmin --strip-components 1

cd phpMyAdmin

sudo mkdir tmp
sudo chown ec2-user:apache tmp
chmod 775 tmp/

sudo rm -rf setup/
sudo cp config.sample.inc.php config.inc.php

sudo nano config.inc.php 
sudo nano /etc/httpd/conf.d/phpMyAdmin.conf
```

**Comment by crisps-for-breakfast** [comment]
I've just updated to phpMyAdmin 5.2.1 and something has changed

SELECT SLEEP(0.5) now just shows the 'Loading' spinner and the page doesn't change, but in the network pane of my developer tools I see `index.php?route=/import` returns  

`TypeError: mysqli::real_connect(): Argument #5 ($port) must be of type ?int, string given`

**Comment by crisps-for-breakfast** [comment]
I tried phpMyAdmin  6.0 and had the same issue as 5.2.1

Version phpMyAdmin 4.9.11 is chock full of php deprecation warnings, but does show a different flavour of the same error:

SELECT SLEEP(0.5) shows:
<img width="1673" alt="image" src="https://user-images.githubusercontent.com/43699463/221003911-41455c40-5714-4e3e-b68c-1a40ea616374.png">

**Comment by crisps-for-breakfast** [comment]
OK, so setting `$cfg['Servers'][$i]['port']` in `/usr/share/config.inc.php` to any truey integer (not 0) seems to work and fix all the issues, at least on 5.2.1

I previously didn't have a line setting that; in fact the only thing I changed was `$cfg['blowfish_secret']`

This feels like a bug, as far as I'm aware if we are connecting via UNIX socket, the port isn't used?

Replacing `$server['port']` with `$server['port'] ?: null` in the two `$mysqli->real_connect()` calls in the DbiMysqli.php class also fixes the problem

**Comment by williamdes** [comment]
Hi @crisps-for-breakfast 
Can you share your `config.inc.php` file ?

**Comment by crisps-for-breakfast** [comment]
Hi @williamdes, sorry forgot to mention that the only `config.inc.php` change I made from `config.sample.inc.php` initially was to fill in a value for

    $cfg['blowfish_secret'] = ''; /* YOU MUST FILL IN THIS FOR COOKIE AUTH! */

**Comment by williamdes** [comment]
Thank you, I keep this one on my list to try to reproduce it

**Comment by williamdes** [comment]
Hi @crisps-for-breakfast
You did state you changed nothing more from the sample config, but did you setup a socket login?
Can you share the complete file here, just scrubbing the sentitive parts?
I can not make sense of how this could fail on the port if you did not change it :)

**Comment by crisps-for-breakfast** [comment]
Thanks @williamdes

I would just be copy and pasting the repo `config.sample.inc` config into here. I've checked again, and the only thing I changed initially was the blowfish secret - which I don't want to share. I don't remember setting up a socket login, but I was under the impression that a socket login was the default when the host is set to 'localhost'

As to how it fails; it looks like `$cfg['Servers'][$i]['port']` is defaulted to `''` or `'0'` when it is not set in `config.sample.inc`. These are `string`s and as such not accepted by `mysqli::real_connect()` which expects `?int`. On reflection, this may only be apparent in newer versions of PHP with strict error reporting, but still represents an issue

Setting `$cfg['Servers'][$i]['port']` to a truthy `int` like `3306` fixes the issue, and is the solution that I have in place

**Comment by kamil-tekiela** [comment]
I don't think this can be reproduced on 6.x anymore. Since 2022 there is `(int)` cast on port. If you want to fix this in 5.2, you can add the same cast in DbiMysqli lines 116 and 126.

**Comment by williamdes** [comment]
Well, after some investigation I decided we are not going to fix this.
Because the line https://github.com/phpmyadmin/phpmyadmin/blame/1da6aeb6c6df1f3cb688b1f30718fc3e8cabaca7/libraries/classes/Config.php#L1345

Already does the work to clean the variable, but more investigation on how this check is bypassed would be welcome @crisps-for-breakfast

**Comment by crisps-for-breakfast** [comment]
Hi William, yep, looks like that line will fix this issue - no idea how it was getting through that. It's been a while since I looked at it to be honest, but I'll re-open if I can recreate it in the future

Thanks for looking :)
